docker/Dockerfile

```diff
@@ -1,25 +1,32 @@
+# https://github.com/rust-lang/docker-rust-nightly/blob/master/buster/Dockerfile
+FROM buildpack-deps:stable-curl AS builder
+
 # Install Rust nightly
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH
+RUN apt update
 COPY docker/install_rust.sh .
 RUN bash ./install_rust.sh
 
 # Install dev tools
-RUN apt-get update && apt-get install -y valgrind
+RUN apt install -y valgrind
 
 # First, fetch and build all the dependencies by compiling an empty
 # crate that depends on everything
-COPY rust/Cargo.lock .
-COPY rust/Cargo.toml .
+COPY rust/Cargo.lock /
+COPY rust/Cargo.toml /
 RUN cargo fetch
-RUN mkdir -p ./src && \
-    touch src/lib.rs && \
+RUN mkdir -p /src && \
+    touch /src/lib.rs && \
     cargo build --lib --all-features
 
 # Now copy the actual source code
-COPY rust/src src
+COPY rust/src /src
 # Fix timestamp. cargo incremental build as issues caused by the above
 # trick to compile the deps separately
-RUN touch src/lib.rs
+RUN touch /src/lib.rs
 RUN cargo build --all-features
+
+FROM alpine:latest
```

Comment:
Is this gonna work? alpine is based on musl, while the builder is based on glibc.

Reply:
I went with the lightest ones for the sake of keeping the environment as light as possible.
The binary that the rust coordinator outputs is dynamically linked or statically linked? If it's static then it _shouldn't_ be a problem, right?

I can also change the `alpine` for another debian-based image that keeps everything light, no problem.

Reply:
Rust binaries dynamically link against the system's libc library.
There is a way to create binaries statically linked with musl, so these binaries can run on any Linux, but afaik that's not the default.

Reply:
See: https://doc.rust-lang.org/edition-guide/rust-2018/platform-and-target-support/musl-support-for-fully-static-binaries.html

Reply:
I'm not against producing musl binaries for docker though, so feel free to go that direction if you have a preference for alpine based containers.

Reply:
Ok, I will weight pros and cons of either going with non-alpine image vs. alpine with statically linked binary.